syntax = "proto2";

package dnp3;

message LinkLayer {
    required uint32 start_bytes = 1 [default = 0x6405];
    required uint32 length = 2 ;
    required uint32 control = 3;
    required uint32 destination_address = 4;
    required uint32 source_address = 5;
    required uint32 crc = 6;
    required bytes  data = 7; // Application layer data
}

message DNP3Packet {
    required LinkLayer link_layer = 1; // Link layer
}

